src/deepsparse/loggers/metric_functions/helpers/config_generation.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Helper functions for generating metric function configs
+"""
+
+__all__ = ["data_logging_config_from_predefined"]
+
+import textwrap
+from typing import Any, Dict, List, Optional, Union
+
+from deepsparse.loggers.config import MetricFunctionConfig
+
+
+_WHITESPACE = "  "
+
+
+def data_logging_config_from_predefined(
+    group_names: Union[str, List[str]],
+    loggers: Dict[str, Optional[Dict[str, Any]]] = {"python": {}},
+    frequency: int = 1,
+    save_dir: Optional[str] = None,
+    save_name: str = "data_logging_config.yaml",
+) -> str:
+    """
+    Generate a data logging config yaml string using a predefined
+    function groups configuration.
+
+    :param group_names: A single group name or a list of group names,
+        that are to be translated into the yaml configuration.
+    :param loggers: Defines set of loggers that will be used to collect
+        the data logs. It is dictionary that maps the logger integration
+        names to their initialization arguments
+    :param frequency: Optional frequency of the data logging
+        functions in the resulting yaml configuration. By default,
+        set to 1
+    :param save_dir: If provided, the resulting yaml configuration is
+        saved to the provided directory
+    :param save_name: If config is saved, it will be under this
+        filename
+    :return: A string yaml dict that specifies the data logging
+        configuration
+    """
+    raise NotImplementedError()
+
+
+def _loggers_to_config_string(
+    loggers: Dict[str, Optional[Union[str, List[str]]]]
+) -> str:
+    return ("\n").join(_nested_dict_to_lines(loggers, yaml_str_lines=[]))
+
+
+def _nested_dict_to_lines(
+    value: Any, key=None, yaml_str_lines=[], level=0
```

Comment:
I know @bfineran , using levels once again :)
Was easier for me to lay out this helper function this way. I will refactor this function to remove this arg before landing ok?

Reply:
level here is fine - the difference is the scope is limited to a single function with self contained logic in its recursion, can be more specific if you want and rename to `_level` to indicate top level users shouldn't specify